data/datasets/sr.py

```diff
@@ -21,10 +21,16 @@ def makedirs(path):
             raise e
 
 
-def load_sr_graph_dataset(name, root=os.path.join(ROOT_DIR, 'datasets')):
+def load_sr_graph_dataset(name, root=os.path.join(ROOT_DIR, 'datasets'), prefer_pkl=False):
     raw_dir = os.path.join(root, 'SR_graphs', 'raw')
     load_from = os.path.join(raw_dir, '{}.g6'.format(name))
-    data = load_sr_dataset(load_from)
+    load_from_pkl = os.path.join(raw_dir, '{}.pkl'.format(name))
+    if prefer_pkl and osp.exists(load_from_pkl):
```

Comment:
In essence, in order to perform the permutation tests I first generate another raw dataset with permuted raw SR _graphs_. It was much more convenient for me to write those in pickle, rather than the original g6 format we have for them. This means that those permuted graphs will need to be loaded from a pickle format rather than g6. I thought it was good to have a flexible behaviour and fall back on g6 given we have both the two versions. If we force one specific loading option, then we will need to pass the correct arg from outside depending on whether we need to load permuted or non permute ones, so it felt a bit heavy.

Any other suggestions/thoughts?

Reply:
Makes sense.